image/manifest.go

```diff
@@ -75,3 +76,29 @@ func manifestLayerInfosToBlobInfos(layers []manifest.LayerInfo) []types.BlobInfo
 	}
 	return blobs
 }
+
+// manifestConvertFn is used to encapsulate helper manifest converstion functions
+// to perform applying of manifest update information.
+type manifestConvertFn func(context.Context, types.ManifestUpdateInformation) (types.Image, error)
+
+// convertManifestIfRequired will run conversion functions of a manifest if
+// required and re-apply the options to the converted type.
```

Comment:
```suggestion
// required and re-apply the options to the converted type.
// It returns (nil, nil) if no conversion was requested.
```
(Absolutely non-blocking.)